File: 5longestPalindromicSubstring/longestPalindromicSubstring.java
class Solution {
    public String longestPalindrome(String s) {
        final var replace = "$#" + java.util.regex.Pattern.compile("\\w").matcher(s).replaceAll(group -> group.group() + "#");
        var center = 0;
        var right = 0;
        final var radius = new int[replace.codePointCount(0, replace.length())];
        for (final var $: (Iterable<Integer>)IntStream.range(1, radius.length)::iterator)
        {
            radius[$] = right > $ ? Math.min(radius[2 * center - $], right - $) : 1;
            while ($ + radius[$] < radius.length && replace.codePointAt($ + radius[$]) == replace.codePointAt($ - radius[$])) ++radius[$];
            if ($ + radius[$] > right)
            {
                center = $;
                right = $ + radius[$];
            }
        }
        center = Arrays.stream(radius).boxed().collect(Collectors.toList()).indexOf(Arrays.stream(radius).max().orElse(0));
        return s.substring((center - radius[center]) / 2, (center + radius[center]) / 2 - 1);
    }
}

File: 5longestPalindromicSubstring/longestPalindromicSubstring.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
    const replace = [...'$#', ...s.replace(/\w/g, group => group + '#')]
    let center = 0, right = 0
    const radius = [...Array(replace.length).fill(0)]
    for (let _ = 1; !Object.is(_, replace.length); ++_)
    {
        radius[_] = right > _ ? Math.min(radius[2 * center - _], right - _) : 1
        while (replace[_ + radius[_]] == replace[_ - radius[_]]) ++radius[_]
        if (_ + radius[_] > right)
        {
            center = _
            right = _ + radius[_]
        }
    }
    center = radius.indexOf(Math.max(...radius))
    return s.slice((center - radius[center]) / 2, (center + radius[center]) / 2 - 1)
}
